prov/gni/include/gnix_freelist.h

```diff
@@ -123,6 +126,12 @@ static inline int _gnix_fl_alloc(struct dlist_entry **e, struct gnix_freelist *f
         fastlock_acquire(&fl->lock);
 
     if (dlist_empty(&fl->freelist)) {
+
+        if (fl->refill_size == 0) {
+            ret = -FI_ECANCELED;
```

Comment:
I'm leaving this one up to Howard. I would like a info or debug message so that we can diagnose any issues in the future.